dll/win32/shlwapi/utils.cpp

```diff
@@ -140,16 +140,35 @@ SHLWAPI_IsBogusHRESULT(HRESULT hr)
     return (hr == E_FAIL || hr == E_INVALIDARG || hr == E_NOTIMPL);
 }
 
+// Used for IShellFolder_GetDisplayNameOf
+struct RETRY_DATA
+{
+    UINT uRemove;
+    UINT uAdd;
+    DWORD dwRetryFlags;
+};
+static const RETRY_DATA g_RetryData[] =
+{
+    { SHGDN_FOREDITING,    0,                SFGDNO_RETRYALWAYS         },
+    { SHGDN_FORADDRESSBAR, 0,                SFGDNO_RETRYALWAYS         },
+    { 0,                   SHGDN_FORPARSING, SFGDNO_RETRYALWAYS         },
+    { SHGDN_FORPARSING,    0,                SFGDNO_RETRYWITHFORPARSING },
+    { SHGDN_INFOLDER,      0,                SFGDNO_RETRYALWAYS         },
+};
+
 /*************************************************************************
  * IShellFolder_GetDisplayNameOf [SHLWAPI.316]
+ *
+ * @note Don't confuse with <shobjidl.h> inline function of the same name.
+ *       This function retries with other flags.
```

Comment:
Looking a bit at the function again, I think you could modify the second sentence to something like this:
`If dwRetryFlags is set and the original call fails with the given uFlags, this function will retry with other flags to attempt retrieving any meaningful description.`
Maybe @cbialorucki or @DarkFire01 could advise on a more understandable sentence than mine.

Reply:
Fixed.